siliconcompiler/tools/klayout/klayout_show.py

```diff
@@ -64,17 +66,129 @@ def show(schema, tech, input_path, output_path, screenshot=False):
 
     # If 'screenshot' mode is set, save image and exit.
     if screenshot:
-        # Save a screenshot. TODO: Get aspect ratio from sc_cfg?
-        horizontal_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var',
-                                               'show_horizontal_resolution',
-                                               step=step, index=index)[0])
-        vertical_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var',
-                                             'show_vertical_resolution',
-                                             step=step, index=index)[0])
+        xbins = int(schema.get('tool', 'klayout', 'task', task, 'var', 'xbins',
+                               step=step, index=index)[0])
+        ybins = int(schema.get('tool', 'klayout', 'task', task, 'var', 'ybins',
+                               step=step, index=index)[0])
 
-        gds_img = layout_view.get_image(horizontal_resolution, vertical_resolution)
-        print(f'[INFO] Saving screenshot to {output_path}')
-        gds_img.save(output_path, 'PNG')
+        if xbins == 1 and ybins == 1:
+            __screenshot(schema, layout_view, output_path)
+        else:
+            __screenshot_montage(schema, layout_view, xbins, ybins)
+
+
+def __screenshot(schema, layout_view, output_path):
+    flow = schema.get('option', 'flow')
+    step = schema.get('arg', 'step')
+    index = schema.get('arg', 'index')
+    task = schema.get('flowgraph', flow, step, index, 'task')
+
+    # Save a screenshot. TODO: Get aspect ratio from sc_cfg?
+    horizontal_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var',
+                                           'show_horizontal_resolution',
+                                           step=step, index=index)[0])
+    vertical_resolution = int(schema.get('tool', 'klayout', 'task', task, 'var',
+                                         'show_vertical_resolution',
+                                         step=step, index=index)[0])
+
+    gds_img = layout_view.get_image(horizontal_resolution, vertical_resolution)
+    print(f'[INFO] Saving screenshot to {output_path}')
+    gds_img.save(output_path, 'PNG')
+
+
+def __screenshot_montage(schema, view, xbins, ybins):
+    flow = schema.get('option', 'flow')
+    step = schema.get('arg', 'step')
+    index = schema.get('arg', 'index')
+    task = schema.get('flowgraph', flow, step, index, 'task')
+
+    app = pya.Application.instance()
+    options = (
+        "grid-show-ruler",
+        "grid-visible",
+        "text-visible"
+    )
+    for option in options:
+        app.set_config(option, "false")
+
+    app.set_config("background-color", "#000000")  # Black
```

Comment:
Any particular reason to make these aesthetic tweaks for the tiled screenshot but not the regular one?

Reply:
the text, grid, and ruler don't stich together well, so they are best to turn off. Maybe one day we can make some options for them